torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -294,100 +234,194 @@ def _wrap_fx_args_as_onnxscript_args(
+                onnxscript_graph_builder.TorchScriptTensor,
+                Tuple[onnxscript_graph_builder.TorchScriptTensor, ...],
+            ],
+        ] = {}
+        self.tracer = None
+        self.op_level_debug = op_level_debug
+        self.diagnostic_context = diagnostic_context
+
+        # TODO: Refactor where/how onnxscript_graph is store/passed in
+        self.tracer = onnxscript_graph_builder.TorchScriptTracingEvaluator(
+            self.onnxscript_graph
         )
-        location = _location_from_fx_stack_trace(node_stack_trace)
-        if location is not None:
-            diagnostic.with_location(location)
 
-    if node.op == "placeholder":
+    @_beartype.beartype
+    def run(self):
+        """Analyze the specified FX node and trigger the ONNX translation for it.
+
+        Args:
+            node: The TorchFX node to dispatch the function for.
+            onnx_args: The arguments of the ONNX function.
+            onnx_kwargs: The keyword arguments of the ONNX function.
+            diagnostic_context: The diagnostic context to use for reporting errors.
+
+        Returns:
+            Either an `onnxscript.OnnxFunction` or `onnxscript.TracedOnnxFunction` instance based on the dispatch algorithm.
+
+        Raises:
+            RuntimeError: If there are no overloaded functions available for the given FX node.
+        """
+
+        # TODO: Fix FakeTensorMode limitation asap
+        # We want to pass list of ints and floats to TorchScript graph correctly
+        # in _export_fx_to_ts, so we must disable FakeTensorMode. Otherwise, graph may
+        # receive FakeTensor and results runtime error. In addition, TorchScript-based
+        # ONNX exporter used in _ts_graph_to_onnx_model_in_protobuf is not compatible
+        # with FakeTensorMode.
+        with torch.utils._mode_utils.no_dispatch():
+            # node_fixed_shape is only used on op_level_debug purpose.
+            for node in self.graph_module.graph.nodes:
+                # Dispatch to ONNX op through OpSchema. The input argument dtypes are compared to
+                # function signature in OpSchema, and find the best matched overload.
+                # TODO(titaiwang): diagnostic rules.
+
+                if node.op == "placeholder":
+                    self.placeholder(node)
+                elif node.op == "get_attr":
+                    self.get_attr(node)
+                elif node.op == "call_function":
+                    self.call_function(node)
+                elif node.op == "call_method":
+                    self.call_method(node)
+                elif node.op == "call_module":
+                    self.call_module(node)
+                elif node.op == "output":
+                    self.output(node)
+                else:
+                    raise RuntimeError(
+                        f"Found node type not defined in torch.fx: {node.op}"
+                    )
+
+        # TODO (thiagofc) Not sure what to do with this
```

Comment:
Original call: https://github.com/pytorch/pytorch/blob/main/torch/onnx/_internal/fx/passes/fx_to_onnxscript.py#L317-L328

Reassigned within `call_function` handling at https://github.com/pytorch/pytorch/blob/main/torch/onnx/_internal/fx/passes/fx_to_onnxscript.py#L424, which also seems to setup diagnostic, but not actually use it?

Is `diagnostic` being used at all in the original, and this PR as a consequence, at all?

Reply:
The code retrieves `inflight_diagnostic` and updates it. So no, not dead code.

Reply:
Nice to hear. I probably will need help to fix this (actually understand how it works).
There is a lot of implied requirement in using `inflight_diagnostic` which resulted in odd issues